.golangci.yaml

```diff
@@ -0,0 +1,161 @@
+    # Default: true
+    http-method: true
+    # Suggest the use of http.StatusXX.
+    # Default: true
+    http-status-code: true
+    # Suggest the use of time.Weekday.String().
+    # Default: true
+    # We don't want this
+    time-weekday: false
+    # Suggest the use of constants available in time package
+    # Default: false
+    time-layout: true
+
+  nolintlint:
+    # Disable to ensure that all nolint directives actually have an effect.
+    # Default: false
+    allow-unused: true  # too many false positive reported
+    # Exclude following linters from requiring an explanation.
+    # Default: []
+    allow-no-explanation: []
+    # Enable to require an explanation of nonzero length
+    # after each nolint directive.
+    # Default: false
+    require-explanation: true
+    # Enable to require nolint directives to mention the specific
+    # linter being suppressed.
+    # Default: false
+    require-specific: true
+
+  # define the import orders
+  gci:
+    sections:
+      # Standard section: captures all standard packages.
+      - standard
+      # Default section: catchall that is not standard or custom
+      - default
+      # Custom section: groups all imports with the specified Prefix.
+      - prefix(github.com/Boeing/config-file-validator)
+
+  staticcheck:
+    # SAxxxx checks in https://staticcheck.io/docs/configuration/options/#checks
+    checks: ["all"]
+
+  revive:
+    enable-all-rules: true
+    rules:
+      # we must provide configuration for linter that requires them
+      # enable-all-rules is OK, but many revive linters expect configuration
+      # and cannot work without them
+
+      # https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#context-as-argument
+      - name: context-as-argument
+        arguments:
+          - allowTypesBefore: "*testing.T"
+
+      # https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#exported
+      - name: exported
+        arguments:
+          # enables checking public methods of private types
+          - "checkPrivateReceivers"
+          # make error messages clearer
+          - "sayRepetitiveInsteadOfStutters"
+
+      # https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#unhandled-error
+      - name: unhandled-error
+        arguments: # here are the exceptions we don't want to be reported
+          - "fmt.Print.*"
+          - "fmt.Fprint.*"
+          - "bytes.Buffer.Write"
+          - "bytes.Buffer.WriteByte"
+          - "bytes.Buffer.WriteString"
+          - "strings.Builder.WriteString"
+          - "strings.Builder.WriteRune"
+
+      # disable everything we don't want
+      - name: add-constant
+        disabled: true # too noisy
+      - name: line-length-limit
+        disabled: true
+      - name: argument-limit
+        disabled: true
+      - name: cognitive-complexity
+        disabled: true
+      - name: banned-characters
+        disabled: true
+      - name: cyclomatic
```

Comment:
Any reason to turn this one off? Same with congnitive-complexity?

Reply:
I decide to keep the out of the first golangci-lint config, but they are indeed interesting.

I would recommend you to merge, then to play with them.